pkg/providers/equinix.go

```diff
@@ -104,6 +104,11 @@ func (p *equinixProvider) AcquireCompleted(id string) (bool, Resource, error) {
 
 	device, _, err := p.client.Devices.Get(id, nil)
 	if err != nil {
+		// "404 Not found" happens when Equinix deletes a failed device
+		// the error is permanent
+		if strings.Contains(err.Error(), "404 Not found") {
+			return false, resource, fmt.Errorf("%w: %s", NewResourceNotFoundError(id), err)
```

Comment:
I'm wondering if `ResourceNotFoudError` it's the right error type for capturing this specific event, or if it could be better to decline it with something more generic like `AcquisitionError`, and letting it wrap the original err